src/Logic.lidr

```diff
@@ -0,0 +1,1296 @@
+λΠ> :t Plus_fact
+Plus_fact : Type
+```
+
+We can later use this name in any situation where a proposition is expected --
+for example, in a function declaration.
+
+> plus_fact_is_true : Plus_fact
+> plus_fact_is_true = Refl
+
+(Here's the reason - recall that names starting with lowercase letters are
+considered implicits in Idris, so \idr{plus_fact} would be considered a free
+variable!)
+
+We can also write _parameterized_ propositions -- that is, functions that take
+arguments of some type and return a proposition. For instance, the following
+function takes a number and returns a proposition asserting that this number is
+equal to three:
+
+> is_three : Nat -> Type 
+> is_three n = n=3
+
+```idris
+λΠ> :t is_three
+is_three : Nat -> Type
+```
+
+In Idris, functions that return propositions are said to define _properties_ of
+their arguments.
+
+For instance, here's a (polymorphic) property defining the familiar notion of an
+_injective function_.
+
+> Injective : (f : a -> b) -> Type
+> Injective {a} {b} f = (x, y : a) -> f x = f y -> x = y
+
+> succ_inj : Injective S
+> succ_inj x x Refl = Refl
+
+The equality operator \idr{=} is also a function that returns a \idr{Type}.
+
+The expression \idr{n = m} is syntactic sugar for \idr{(=) n m}, defined
+internally in Idris. Because \idr{=} can be used with elements of any type, it
+is also polymorphic:
+
+```idris
+λΠ> :t (=)
+(=) : A -> B -> Type
+```
+
+
+== Logical Connectives
+
+=== Conjunction
+
+The _conjunction_ (or _logical and_) of propositions \idr{a} and \idr{b} in
+Idris is the same as the pair of \idr{a} and \idr{b}, written \idr{(a, b)},
+representing the claim that both \idr{a} and \idr{b} are true.
+
+> and_example : (3 + 4 = 7, 2 * 2 = 4)
+
+To prove a conjunction, we can use value-level pair syntax:
+
+> and_example = (Refl, Refl)
+
+For any propositions \idr{a} and \idr{b}, if we assume that \idr{a} is true and
+we assume that \idr{a} is true, we can trivially conclude that \idr{(a,b)} is
+also true.
+
+> and_intro : a -> b -> (a, b)
+> and_intro = MkPair
+
+
+==== Exercise: 2 stars (and_exercise)
+
+> and_exercise : (n, m : Nat) -> n + m = 0 -> (n = 0, m = 0)
+> and_exercise n m prf = ?and_exercise_rhs
+
+$\square$
+
+So much for proving conjunctive statements. To go in the other direction --
+i.e., to _use_ a conjunctive hypothesis to help prove something else -- we
+employ pattern matching.
+
+
+If the proof context contains a hypothesis \idr{h} of the form \idr{(a,b)}, case
+splitting will replace it with a pair pattern \idr{(a,b)}.
+
+> and_example2 : (n, m : Nat) -> (n = 0, m = 0) -> n + m = 0
+> and_example2 Z Z (Refl,Refl) = Refl
+> and_example2 (S _) _ (Refl,_) impossible
+> and_example2 _ (S _) (_,Refl) impossible
+
+You may wonder why we bothered packing the two hypotheses \idr{n = 0} and \idr{m
+= 0} into a single conjunction, since we could have also stated the theorem with
+two separate premises:
+
+> and_example2'' : (n, m : Nat) -> n = 0 -> m = 0 -> n + m = 0
```

Comment:
Ah, it's because I've skipped `and_example2'`due to it being identical to `and_example2` in Idris